scenario/scripts/snapshot.py

```diff
@@ -0,0 +1,752 @@
+import json
+import logging
+import os
+import re
+import tempfile
+from dataclasses import asdict
+from enum import Enum
+from itertools import chain
+from pathlib import Path
+from subprocess import CalledProcessError, check_output, run
+from textwrap import dedent
+from typing import Any, BinaryIO, Dict, Iterable, List, Optional, TextIO, Tuple, Union
+
+import ops.pebble
+import typer
+import yaml
+
+from scenario.state import (
+    Address,
+    BindAddress,
+    Container,
+    Model,
+    Mount,
+    Network,
+    Relation,
+    State,
+    Status,
+)
+
+logger = logging.getLogger("snapshot")
+
+JUJU_RELATION_KEYS = frozenset({"egress-subnets", "ingress-address", "private-address"})
+JUJU_CONFIG_KEYS = frozenset({})
+SNAPSHOT_TEMPDIR_ROOT = (Path(os.getcwd()).parent / "snapshot_storage").absolute()
```

Comment:
The main reason is that if you're fetching mount contents it's 99% of the times because you want to be able to either inspect the files or store them for good so that you can write a unittest that uses them as they are (i.e. reproduce a real charm run).

So saving them to a tempdir that will be cleaned up immediately after the command has printed out their location is pointless, if you're not interested in that data, simply don't fetch it by omitting 'c' from the `includes` arg.

That said, I suppose there is value in customizing that directory, so I added a todo to accept it as a cli argument.

Reply:
Ah okay, generally when I see "TEMP" as part of the name of something I think that it is only going to be kept around during the program's execution and not persist afterwards. But if the files are being used afterwards to produce unit tests, then that would explain why you are not using the `tempfile` module. Maybe the name could be something more clear like `SNAPSHOT_OUTPUT_DIR` or something similar?

Reply:
yes, that's a good idea.